hazelcast-jclouds/src/test/java/com/hazelcast/jclouds/ComputeServiceBuilderInvalidConfigTest.java

```diff
@@ -0,0 +1,139 @@
+    public void test_invalid_config_when_both_IAM_and_credential_set() throws Exception {
+
+        String propertiesUnderTest = "<property name=\"provider\">aws-ec2</property>" +
+                                     "<property name=\"role-name\">test</property>" +
+                                     "<property name=\"identity\">test</property>" +
+                                     "<property name=\"credential\">test</property>";
+
+        String configXML = configTemplate.replace("${PROPERTIES_PLACE_HOLDER}", propertiesUnderTest);
+        Config config = new XmlConfigBuilder(new ByteArrayInputStream(stringToBytes(configXML))).build();
+        Hazelcast.newHazelcastInstance(config);
+    }
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void test_invalid_config_when_IAM_role_is_configured_with_nonEc2_Provider() throws Exception {
+
+        String propertiesUnderTest = "<property name=\"provider\">google-compute-engine</property>" +
+                                      "<property name=\"role-name\">test</property>";
+
+        String configXML = configTemplate.replace("${PROPERTIES_PLACE_HOLDER}", propertiesUnderTest);
+        Config config = new XmlConfigBuilder(new ByteArrayInputStream(stringToBytes(configXML))).build();
+        Hazelcast.newHazelcastInstance(config);
+    }
+
+    @Test(expected = InvalidConfigurationException.class)
+    public void test_invalid_config_when_tag_keys_and_values_have_different_sizes() throws Exception {
+
+        String propertiesUnderTest = "<property name=\"provider\">aws-ec2</property>" +
+                                     "<property name=\"identity\">test</property>" +
+                                     "<property name=\"credential\">test</property>" +
+                                     "<property name=\"tag-keys\">tag1,tag2,tag3</property>" +
+                                     "<property name=\"tag-values\">tag1,tag2</property>";
+
+
+        String configXML = configTemplate.replace("${PROPERTIES_PLACE_HOLDER}", propertiesUnderTest);
+        Config config = new XmlConfigBuilder(new ByteArrayInputStream(stringToBytes(configXML))).build();
+        Hazelcast.newHazelcastInstance(config);
+    }
+
+    @Test(expected = RuntimeException.class)
+      public void test_invalid_config_when_port_config_exceed_max_value() throws Exception {
+
+        String propertiesUnderTest = "<property name=\"provider\">aws-ec2</property>" +
+                "<property name=\"identity\">test</property>" +
+                "<property name=\"credential\">test</property>" +
+                "<property name=\"hz-port\">78000</property>";
+
+        String configXML = configTemplate.replace("${PROPERTIES_PLACE_HOLDER}", propertiesUnderTest);
+        Config config = new XmlConfigBuilder(new ByteArrayInputStream(stringToBytes(configXML))).build();
+        Hazelcast.newHazelcastInstance(config);
+    }
+
+    @Test(expected = RuntimeException.class)
```

Comment:
Because our `DefaultDiscoveryService`  impl converts all config exceptions to `RuntimeException` here : https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/spi/discovery/impl/DefaultDiscoveryService.java#L150